src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ManagedStrategy.kt

```diff
@@ -1356,12 +1359,18 @@ abstract class ManagedStrategy(
     }
 
     private fun methodGuaranteeType(owner: Any?, className: String, methodName: String): ManagedGuaranteeType? = runInIgnoredSection {
-        userDefinedGuarantees?.forEach { guarantee ->
-            val ownerName = owner?.javaClass?.canonicalName ?: className
-            if (guarantee.classPredicate(ownerName) && guarantee.methodPredicate(methodName)) {
-                return guarantee.type
+        predefinedGuarantees
```

Comment:
Let's not re-use `ManagedStrategyGuarantee` mechanism for this purpose. 

1. I remember there were some performance problem, because the lookup through the list of guarantees was working slowly. 
2. We might revisit their API at some point.

So could you just hard-code here handling of I/O streams?

In the long run, we need to revisit this place in the code anyway, because e.g., I think we also need to return `Ignored` guarantee for classes for which `shouldTransform` returns false. 
So we need some centralized mechanism to specify guarantees of different classes/methods.